charset/charset.go

```diff
@@ -57,15 +57,18 @@ var charsetInfos = map[string]*Charset{
 	CharsetASCII:   {CharsetASCII, CollationASCII, make(map[string]*Collation), "US ASCII", 1},
 	CharsetLatin1:  {CharsetLatin1, CollationLatin1, make(map[string]*Collation), "Latin1", 1},
 	CharsetBin:     {CharsetBin, CollationBin, make(map[string]*Collation), "binary", 1},
+	CharsetGBK:     {CharsetGBK, CollationGBKChineseCI, make(map[string]*Collation), "GBK", 2},
 }
 
 // All the names supported collations should be in the following table.
 var supportedCollationNames = map[string]struct{}{
-	CollationUTF8:    {},
-	CollationUTF8MB4: {},
-	CollationASCII:   {},
-	CollationLatin1:  {},
-	CollationBin:     {},
+	CollationUTF8:          {},
+	CollationUTF8MB4:       {},
+	CollationASCII:         {},
+	CollationLatin1:        {},
+	CollationBin:           {},
+	CollationGBKChineseCI:  {},
```

Comment:
`supportedCollationNames` is not expected to change, it is used for `show collations` when new collation is not enabled.